core/commands/generate_command.rb

```diff
@@ -211,33 +213,58 @@ def make_product_config(product_name, product, box, repo)
     attribute_name = @env.repos.attribute_name(product_name)
     { "#{attribute_name}": config }
   end
+  # rubocop:enable Metrics/CyclomaticComplexity
 
   # Make the list of the role parameters in the JSON-format.
   #
   # @param name [String] internal name of the machine specified in the template
   # @param product_config [Hash] list of the product parameters
   # @param recipe_name [String] name of the recipe
+  # @param subscription_manager_params [Hash] subscription manager params
+  # in format { recipe_name, credentials, attribute_name }
   # @return [String] pretty formatted role description in JSON format.
-  def make_role_json(name, product_config, recipe_name)
-    role = {
-      name: name,
-      default_attributes: {},
-      override_attributes: product_config,
-      json_class: 'Chef::Role',
-      description: '',
-      chef_type: 'role',
-      run_list: ['recipe[mdbci_provision_mark::remove_mark]',
-                 "recipe[#{recipe_name}]",
-                 'recipe[mdbci_provision_mark::default]']
-    }
+  def make_role_json(name, product_config, recipe_name, subscription_manager_params)
+    run_list = ['recipe[mdbci_provision_mark::remove_mark]',
+                "recipe[#{recipe_name}]",
+                'recipe[mdbci_provision_mark::default]']
+    unless subscription_manager_params.empty?
+      run_list.insert(1, subscription_manager_params[:recipe_name])
+      product_config = product_config.merge(
+        subscription_manager_params[:attribute_name] => subscription_manager_params[:credentials]
+      )
+    end
+    role = { name: name,
+             default_attributes: {},
+             override_attributes: product_config,
+             json_class: 'Chef::Role',
+             description: '',
+             chef_type: 'role',
+             run_list: run_list }
     JSON.pretty_generate(role)
   end
 
+  # Generate the subscription manager params for generating role JSON-file.
+  #
+  # @param box_params [Hash] information of the box parameters
+  # @param credentials [Hash] information of the box credentials
+  # @return [Hash] subscription manager params in format { recipe_name, credentials, attribute_name }
+  # for generating role JSON-file.
+  # @raise RuntimeError if RHEL credentials for Red Hat Subscription-Manager are not configured.
+  def make_subscription_manager_params(box_params, credentials)
+    return {} unless box_params['configure_subscription_manager'] == 'true'
+
+    raise 'RHEL credentials for Red Hat Subscription-Manager are not configured' if credentials.nil?
+
+    { recipe_name: 'recipe[subscription-manager]',
```

Comment:
These are the constants. Why not just use constants in the `make_role_json` method instead of passing them around as hash parameters?